source/ncconvert/analysis/GeometryAnalysis.cpp

```diff
@@ -0,0 +1,147 @@
+#include "GeometryAnalysis.h"
+#include "ncasset/AssetTypes.h"
+
+#include "ncutility/NcError.h"
+
+#include <algorithm>
+#include <concepts>
+#include <iostream>
+#include <type_traits>
+
+namespace
+{
+auto CompareSquareMagnitudes(const nc::Vector3& lhs, const nc::Vector3& rhs)
+{
+    return nc::SquareMagnitude(lhs) < nc::SquareMagnitude(rhs);
+}
+
+auto PositionProjection(const nc::asset::MeshVertex& vertex) -> const nc::Vector3&
```

Comment:
Why is this called PositionProjection?

Reply:
The standard algorithms in the `ranges` namespace have an optional projection function that transforms input data before passing to the actual operation, and this guy is used for those algorithms. All of the mesh analysis stuff only cares about positional data, so we 'project' the `MeshVertex` to just its position. Added a comment here.